src/Franz/Brokers.fs

```
@@ -65,8 +65,12 @@ type Broker(nodeId : Id, endPoint : EndPoint, leaderFor : TopicPartitionLeader a
             try
                 send self request
             with
+            | :? InvalidOperationException as e ->
```

Maybe we should throw a custom exception when connection is broken, just to make sure it's the expected scenario we are handling. InvalidOperationException seems a bit generic in this case